src/builders.ts

```diff
@@ -771,3 +773,82 @@ export function makeContractNonFungiblePostCondition(
     assetName
   );
 }
+
+/**
+ * Read only function response object
+ *
+ * @param {Boolean} okay - the status of the response
+ * @param {string} result - serialized hex clarity value
+ */
+
+export interface ReadOnlyFunctionResponse {
+  okay: boolean;
+  result: string;
+}
+
+/**
+ * Read only function options
+ *
+ * @param  {String} contractAddress - the c32check address of the contract
+ * @param  {String} contractName - the contract name
+ * @param  {String} functionName - name of the function to be called
+ * @param  {[ClarityValue]} functionArgs - an array of Clarity values as arguments to the function call
+ * @param  {StacksNetwork} network - the Stacks blockchain network this transaction is destined for
+ * @param  {String} senderAddress - the c32check address of the sender
+ */
+
+export interface ReadOnlyFunctionOptions {
+  contractName: string;
+  contractAddress: string;
+  functionName: string;
+  functionArgs: ClarityValue[];
+  network?: StacksNetwork;
+  senderAddress: string;
+}
+
+/**
+ * Calls a read only function from a contract interface
+ *
+ * @param  {ReadOnlyFunctionOptions} readOnlyFunctionOptions - the options object
+ *
+ * Returns an object with a status bool (okay) and a result string that is a serialized clarity value in hex format.
+ *
+ * @return {ReadOnlyFunctionResponse}
+ */
+export async function callReadOnlyFunction(
+  readOnlyFunctionOptions: ReadOnlyFunctionOptions
+): Promise<ReadOnlyFunctionResponse> {
+  const defaultOptions = {
+    network: new StacksMainnet(),
+  };
+
+  const options = Object.assign(defaultOptions, readOnlyFunctionOptions);
+
+  const {
+    contractName,
+    contractAddress,
+    functionName,
+    functionArgs,
+    network,
+    senderAddress,
+  } = options;
+
+  const url = `${network.coreApiUrl}/v2/contracts/call-read/${contractAddress}/${contractName}/${functionName}`;
```

Comment:
function names can contain `?` and should be encoded with `encodeURIComponent`

Reply:
Oh good to know! Do you have a contract that I could test with?

Reply:
Something like this?

```
(define-read-only (info?)
  (some u1)
)
```